frontend/javascripts/oxalis/geometries/materials/plane_material_factory.ts

```diff
@@ -971,11 +971,12 @@ class PlaneMaterialFactory {
       names,
       ({ isSegmentationLayer }) => !isSegmentationLayer,
     ).map((layers) => layers.map(({ name }) => sanitizeName(name)));
+    const colorNameSet = new Set(sanitizedColorLayerNames);
 
     return [
       sanitizedColorLayerNames,
       sanitizedSegmentationLayerNames,
-      sanitizedOrderedColorLayerNames,
+      allSanitizedOrderedColorLayerNames.filter((name) => colorNameSet.has(name)),
```

Comment:
Does this ensure that `allSanitizedOrderedColorLayerNames` does not include null values and names of segmentation layer?

If that's the case this sure fixes the bug, but does not tackle the actual root cause of the problem. Imo `allSanitizedOrderedColorLayerNames` should only have valid sanitized color layer names in a specific order in it. If that's not the case here, maybe something around line 925 should be changed :thinking: 

If that's not the case, I do not understand what causes the bug.

Reply:
The problem is that often not all layers can be rendered. In that case, only a subset of layers is rendered (this method is responsible for choosing these). `sanitizedColorLayerNames` only contains the layers that should be rendered, but `sanitizedOrderedColorLayerNames` contained more in the error case. However, the arrays should be the same length. Due to this mismatch, the shader tried to refer to some layers that the shader shouldn't know of.
Maybe we overhaul the rendering approach at the next hackathon. So, I just merged the PR now :)